src/Mutator/Boolean/LogicalOr.php

```diff
@@ -77,6 +79,76 @@ public function mutate(Node $node): iterable
 
     public function canMutate(Node $node): bool
     {
-        return $node instanceof Node\Expr\BinaryOp\BooleanOr;
+        if (!$node instanceof Node\Expr\BinaryOp\BooleanOr) {
+            return false;
+        }
+
+        $equalOp = [
+            Node\Expr\BinaryOp\Identical::class,
+            Node\Expr\BinaryOp\Equal::class,
+        ];
+
+        if (
+            in_array(get_class($node->left), $equalOp, true) === true
+            && in_array(get_class($node->right), $equalOp, true) === true
+        ) {
+            $varNameLeft = null;
+
+            if ($node->left->left instanceof Node\Expr\Variable) {
+                $varNameLeft = $node->left->left->name;
+            } elseif ($node->left->right instanceof Node\Expr\Variable) {
```

Comment:
could you please explain what source code leads to this checks/AST? I think I don't understand AST structure :)

Reply:
😅 
```php
$myVar === 'hello' || $myVar === 'world';
```

for example (see added tests). Because it must not be mutated to 

```php
$myVar === 'hello' && $myVar === 'world';
```

So I check that on both side of the `||` we compare the same variable name against an equality. but because the variable could be on either the right side of `===` or the left side, I have to check node->left->left, node->left->right + node->right->right, node->left->left  for the variable name.

Reply:
cool, now I fully understand it. Then IMO in tests we need to add a case where on the left side of `||` we have a variable on the right side of `===` (now there is no such case)

Reply:
yes there is : https://github.com/infection/infection/pull/1801/files#diff-6cb62609cbceaf882b2803a9e26b277dc505d2cae4258e29d23acf9a9d067d37R106

Reply:
I mean on the left side of `||`. So additional test will be

```diff
+ 'hello' === $myVar || 'world' === $myVar;
```

Reply:
fixed.